pygfx/geometries/_cylinder.py

```diff
@@ -134,62 +133,82 @@ def generate_cap(radius, height, radial_segments, theta_start, theta_length, up=
-            groups = [mesh]
-            if radius_bottom > 0:
-                bottom_cap = generate_cap(
-                    radius_bottom,
-                    -height / 2,
-                    radial_segments,
-                    theta_start,
-                    theta_length,
-                    up=False,
-                )
-                groups.append(bottom_cap)
-            if radius_top > 0:
-                top_cap = generate_cap(
-                    radius_top,
-                    height / 2,
-                    radial_segments,
-                    theta_start,
-                    theta_length,
-                    up=True,
-                )
-                groups.append(top_cap)
-            mesh = merge(groups)
-
-        positions, normals, texcoords, indices = mesh
-        self.positions = Buffer(positions)
-        self.normals = Buffer(normals)
-        self.texcoords = Buffer(texcoords)
-        self.indices = Buffer(indices.reshape((-1, 3)))
+def cylinder_geometry(
+    radius_bottom=1.0,
+    radius_top=1.0,
+    height=1.0,
+    radial_segments=8,
+    height_segments=1,
+    theta_start=0.0,
+    theta_length=np.pi * 2,
+    open_ended=False,
+):
+    """Create geometry representing a cylinder."""
+
+    assert radial_segments > 0
+    assert height_segments > 0
+    assert theta_length != 0.0
+
+    mesh = generate_torso(
+        radius_bottom,
+        radius_top,
+        height,
+        radial_segments,
+        height_segments,
+        theta_start,
+        theta_length,
+    )
+
+    if not open_ended:
+        groups = [mesh]
+        if radius_bottom > 0:
+            bottom_cap = generate_cap(
+                radius_bottom,
+                -height / 2,
+                radial_segments,
+                theta_start,
+                theta_length,
+                up=False,
+            )
+            groups.append(bottom_cap)
+        if radius_top > 0:
+            top_cap = generate_cap(
+                radius_top,
+                height / 2,
+                radial_segments,
+                theta_start,
+                theta_length,
+                up=True,
+            )
+            groups.append(top_cap)
+        mesh = merge(groups)
+
+    positions, normals, texcoords, indices = mesh
+    return Geometry(
+        indices=indices.reshape((-1, 3)),
+        positions=positions,
+        normals=normals,
+        texcoords=texcoords,
+    )
+
+
+def cone_geometry(
+    radius=1.0,
+    height=1.0,
+    radial_segments=8,
+    height_segments=1,
+    theta_start=0.0,
+    theta_length=np.pi * 2,
+    open_ended=False,
+):
+    """Create geometry representing a cone."""
+    return cylinder_geometry(
+        radius_bottom=radius,
+        radius_top=0.0,
+        height=height,
+        radial_segments=radial_segments,
+        height_segments=height_segments,
+        theta_start=theta_start,
+        theta_length=theta_length,
+        open_ended=open_ended,
+    )
```

Comment:
`cone_geometry` could also be defined with `functools.partial` which should be much less verbose :)

Reply:
Not sure. How would that work with docstrings? Also some arguments are named differently. I'm not sure what this'd look like, but this could be because I've rarely used `functools.partial` :)  If you have a nice patch though, I am happy to try that out.

Reply:
Good point, I'll look into those questions.

Partial is quite nice because it is about 10x faster at runtime than calling a function wrapped in another function. I assume it's a bit more cleverly handled internally to cpython.

Reply:
> about 10x faster

wow 🤯 ... that I did not know